lib/modules/manager/kustomize/artifacts.ts

```diff
@@ -0,0 +1,217 @@
+    logger.debug(
+      `Not inflating Helm chart for ${dependency.depName} as kustomizeInflateHelmCharts is not enabled and the current version isn't inflated`,
+    );
+    return;
+  }
+
+  if (
+    is.nonEmptyString(currentChartExistingPath) &&
+    is.nonEmptyString(dependency.newVersion)
+  ) {
+    logger.debug(`Deleting previous helm chart: ${currentChartExistingPath}`);
+    await deleteLocalFile(currentChartExistingPath);
+  }
+
+  const versionToPull = dependency.newVersion ?? dependency.currentVersion;
+
+  if (!is.nonEmptyString(versionToPull)) {
+    return;
+  }
+
+  const versionToPullExistingPath = await localExistingChartPath(
+    chartHome,
+    dependency.depName,
+    versionToPull,
+  );
+
+  if (is.nonEmptyString(versionToPullExistingPath)) {
+    logger.debug(
+      `Helm chart ${dependency.depName} version ${versionToPull} already exists at ${versionToPullExistingPath}`,
+    );
+    return;
+  }
+
+  const folderName = `${dependency.depName}-${versionToPull}`;
+  const untarDir = upath.join(chartHome, folderName);
+  const registryUrl = dependency.registryUrls[0];
+  logger.debug(
+    `Pulling helm chart ${dependency.depName} version ${versionToPull} to ${untarDir}`,
+  );
+  const cmd =
+    `helm pull --untar --untardir ${quote(untarDir)} ` +
+    `--version ${quote(versionToPull)} --repo ${quote(registryUrl)} ${dependency.depName}`;
+
+  await exec([cmd], execOptions);
+}
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug(`kustomize.updateArtifacts(${packageFileName})`);
+  const project = parseKustomize(newPackageFileContent);
+  const isUpdateOptionInflateChartArchives =
+    config.postUpdateOptions?.includes('kustomizeInflateHelmCharts') === true;
+
+  if (is.nullOrUndefined(project)) {
+    return [
+      {
+        artifactError: {
+          stderr: 'Failed to parse new package file content',
+        },
+      },
+    ];
+  }
+
+  const chartHome = getSiblingFileName(
+    packageFileName,
+    project.helmGlobals.chartHome,
+  );
+
+  try {
+    const helmToolConstraint: ToolConstraint = {
+      toolName: 'helm',
+      constraint: config.constraints?.helm,
+    };
+
+    const execOptions: ExecOptions = {
+      docker: {},
+      userConfiguredEnv: config.env,
+      extraEnv: generateHelmEnvs(),
+      toolConstraints: [helmToolConstraint],
+    };
+
+    for (const dependency of updatedDeps) {
+      await inflateHelmChart(
+        execOptions,
+        chartHome,
+        dependency,
+        isUpdateOptionInflateChartArchives,
+      );
+    }
+
+    const status = await getRepoStatus();
+    const chartsAddition = status?.not_added ?? [];
+    const chartsDeletion = status?.deleted ?? [];
+
+    const fileChanges: UpdateArtifactsResult[] = [];
+    await writeLocalFile(packageFileName, newPackageFileContent);
```

Comment:
Doesn't make sense to write a file to disk here. What's the intention?

Reply:
Just inexperience with the project, i wasn't sure if the new packagefile content needed to be written to disk here or if it's done somewhere else.

Reply:
It looks like we don't need to write `newPackageFileContent` in the artifacts management.
I'm running https://github.com/renovatebot/renovate/pull/34277/commits/3f057936e99a43b1dc51f071dc6dc226903e1a40 now and the updated `kustomization.yaml` file is included in the PR without me having to write it in artifacts management, see this PR for example: https://github.com/micke/renovate-kustomize-artifacts/pull/8/files#diff-5c38001a592677b709f1e7de7702d683403356ce7824481d04ed5def445663ba

If that's not by design or if you think we should write it anyways then i'll add it back but we certainly don't need to write it before calling helm as we're not using the file to pull the helm charts...